main/lsp/requests/hover.cc

```diff
@@ -37,18 +37,12 @@ HoverTask::HoverTask(const LSPConfiguration &config, MessageId id, std::unique_p
 unique_ptr<ResponseMessage> HoverTask::runRequest(LSPTypecheckerInterface &typechecker) {
     auto response = make_unique<ResponseMessage>("2.0", id, LSPMethod::TextDocumentHover);
 
-    if (typechecker.isStale()) {
-        config.logger->debug("HoverTask running on stale, returning stub result");
-
-        auto clientHoverMarkupKind = config.getClientConfig().clientHoverMarkupKind;
-        string typeString = "stub_answer_for_hover";
-        string docString = "We would have tried to answer this hover request with stale data\n\n";
+    const core::GlobalState &gs = typechecker.state();
 
-        response->result = make_unique<Hover>(formatRubyMarkup(clientHoverMarkupKind, typeString, docString));
-        return response;
+    if (typechecker.isStale()) {
+        config.logger->debug("HoverTask running on stale data");
```

Comment:
Is there a way we can force the implementation of each LSPMethod to declare whether it's ok to be called with a stale state? I'd love to force opt-in, not opt-out.

(If that was already implemented in a previous PR I'd love to be pointed to the implementation.)

Reply:
There's a virtual method on `LSPTask` that defaults to `return false`, and we'll only feed an `LSPStaleTypechecker` 
 if it returns `true`. Currently only `hover` and `codeAction` opt in.

https://github.com/sorbet/sorbet/blob/db69c5c127d136d49f206e846cbb986cc9818f2f/main/lsp/LSPTask.h#L114-L115